### ISSUE TITLE
Task with name 'bundleSolidRelease' not found in project ':matterlib'.

### ISSUE THREAD

**Issue by danielwilson1702** [mirceanis/multi-flavor-lib-demo#2]
Hi could you explain what is happening here? When I checkout the single lib branch, I get an error that reads:
`Task with name 'bundleSolidRelease' not found in project ':matterlib'.`

```
if (android.productFlavors.size() > 0) {
    android.libraryVariants.all { variant ->
        if (variant.name.toLowerCase().contains("debug")) {
            return
        }

        def bundleTask = tasks["bundle${variant.name.capitalize()}"]

        artifacts {
            archives(bundleTask.archivePath) {
                classifier variant.flavorName
                builtBy bundleTask
                name = project.name
            }
        }
    }
}
```

Can't see what I am doing wrong really this is all pretty confusing, I get the same error in my project 😄 I would like to publish build variants for my library.

**Comment by mirceanis** [comment]
I suspect there has been a change in the task names in more recent versions of the android gradle plugin.
The task names now are suffixed with `Aar`.

I'll add an update soon

**Comment by danielwilson1702** [comment]
Thanks yeah I can see that, where on earth is this documented?

If I update the task to `def bundleTask = tasks["bundle${variant.name.capitalize()}" + 'Aar']` it stops complaining but still won't successfully sync and I can't really see what's wrong with it.

Edit: Ok the version of Kotlin needs updating as well. So it seems Android Studio 3.2.1 just bricks this code by forcing you to use the 3.2.1 android gradle plugin.

**Comment by mirceanis** [comment]
I made a few updates to both branches to fix the recent changes.
I'm using the `3.3.0-beta03` plugin in this demo.
I'll update to a stable 3.3.0 when it's released.

I'm not sure where this is documented, I just followed the tasks run during build and picked the one that makes the artifact.

That piece of code you asked about basically does this:
* when there are multiple flavors
* go through all variants
* ignore the debug builds
* get the task that does the bundling
* produce an artifact and mark it as built by that task
